### PR TITLE
create external GKE cluster

### LINKED ISSUE #8035: Extend create external cluster endpoint
**User Story**

Extend create external cluster endpoint to support GKE cluster

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: extend endpoint to support GKE cluster import

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8035

```release-note
Manage GKE Clusters natively. Extend external cluster functionality to import existing GKE cluster.
```

### CODE REVIEW

File: pkg/handler/v2/external_cluster/external_cluster.go [hunk 1]
```diff
@@ -58,46 +58,110 @@ func CreateEndpoint(userInfoGetter provider.UserInfoGetter, projectProvider prov
...
+	}
+	if adminUserInfo.IsAdmin {
+		return privilegedClusterProvider.NewUnsecured(project, newCluster)
 	}
+	userInfo, err := userInfoGetter(ctx, project.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	return clusterProvider.New(userInfo, project, newCluster)
```
Comment (lsviben) [inline pkg/handler/v2/external_cluster/external_cluster.go]: you can re-use this 
```suggestion
return createNewCluster(ctx, userInfoGetter, clusterProvider, privilegedClusterProvider, newCluster, project)
```

File: pkg/handler/v2/external_cluster/external_cluster.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ const (
 	normalType  = "normal"
 )
 
-func CreateEndpoint(userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, clusterProvider provider.ExternalClusterProvider, privilegedClusterProvider provider.PrivilegedExternalClusterProvider, settingsProvider provider.SettingsProvider) endpoint.Endpoint {
+func CreateEndpoint(userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, clusterProvider provider.ExternalClusterProvider, privilegedClusterProvider provider.PrivilegedExternalClusterProvider, settingsProvider provider.SettingsProvider, presetsProvider provider.PresetProvider) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
 		if !AreExternalClustersEnabled(settingsProvider) {
```
Comment (imharshita) [inline pkg/handler/v2/external_cluster/external_cluster.go:52]: How can user enable external cluster?

Comment (zreigz) [reply pkg/handler/v2/external_cluster/external_cluster.go:52]: in global admin settings

**Pull request merged**